Documentation/contributing/development/hive.rst

```diff
@@ -877,4 +877,150 @@ See also the runnable example in `pkg/k8s/resource/example <https://github.com/c
+        registry job.Registry,
+    ) *exampleCell {
+        ex := exampleCell{
+            jobGroup: registry.NewGroup(
+                job.WithLogger(logger),
+                job.WithPprofLabels(pprof.Labels("cell", "example")),
+            ),
+            workChan: make(chan struct{}, 3),
+            trigger:  job.NewTrigger(),
+            logger:   logger,
+        }
+
+        ex.jobGroup.Add(
+            job.OneShot(
+                "sync-on-startup",
+                ex.sync,
+                job.WithRetry(3, workqueue.DefaultControllerRateLimiter()),
+                job.WithShutdown(), // if the retries fail, shutdown the hive
+            ),
+            job.OneShot("daemon", ex.daemon),
+            job.Timer("timer", ex.timer, 5*time.Second, job.WithTrigger(ex.trigger)),
+            job.Observer("observer", ex.observer, stream.FromChannel(ex.workChan)),
+        )
+
+        lifecycle.Append(ex.jobGroup)
+
+        return &ex
+    }
+
+    func (ex *exampleCell) sync(ctx context.Context) error {
+        for i := 0; i < 3; i++ {
+            if err := ex.doSomeWork(); err != nil {
+                return fmt.Errorf("doSomeWork: %w", err)
+            }
+        }
+
+        return nil
+    }
+
+    func (ex *exampleCell) daemon(ctx context.Context) error {
+        for {
+            randomTimeout := time.NewTimer(time.Duration(rand.Intn(3000)) * time.Millisecond)
+            select {
+            case <-ctx.Done():
+                return nil
+
+            case <-randomTimeout.C:
+                ex.doSomeWork()
+            }
+        }
+    }
+
+    func (ex *exampleCell) timer(ctx context.Context) error {
+        if err := ex.doSomeWork(); err != nil {
+            return fmt.Errorf("doSomeWork: %w", err)
+        }
+
+        return nil
+    }
+
+    func (ex *exampleCell) Trigger() {
+        ex.trigger.Trigger()
+    }
+
+    func (ex *exampleCell) observer(ctx context.Context, event struct{}) error {
+        ex.logger.Info("Observed event")
+        return nil
+    }
+
+    func (ex *exampleCell) HeavyLifting() {
+        ex.jobGroup.Add(job.OneShot("long-running-job", func(ctx context.Context) error {
+            for i := 0; i < 1_000_000; i++ {
+                // Do some heavy lifting
+            }
+            return nil
+        }))
+    }
+
+    func (ex *exampleCell) doSomeWork() error {
+        ex.workChan <- struct{}{}
+        return nil
+    }
+
 
+The above example shows a number of use-cases in one cell. We start by requesting the job.Registry
+via the constructor. We can use the registry to create job groups, in most cases one will be enough.
+To this group we can add our jobs in the constructor. Any jobs added in the constructor are queued
+until the lifecycle of our cell starts. The group is added to the lifecycle and manages this 
+internally. Jobs can also be added at runtime which can be handy for dynamic workloads while still
+guaranteeing a clean shutdown.
```

Comment:
Grammar, clarity, use active voice in present tense, avoid "we" and focus instead on a reader's actions ("you" often works)
```suggestion
The preceding example shows a number of use cases in one cell. The cell starts by requesting the job.Registry
by way of the constructor. The registry can create job groups; in most cases, one is enough.
You can add jobs in the constructor to this group. Any jobs added in the constructor are queued
until the lifecycle of the cell starts. The group is added to the lifecycle and manages jobs 
internally. You can also add jobs at runtime, which can be handy for dynamic workloads while still
guaranteeing a clean shutdown.
```